corehq/apps/data_dictionary/views.py

```diff
@@ -143,6 +149,144 @@ def data_dictionary_json(request, domain, case_type_name=None):
     })
 
 
+@login_and_domain_required
+@requires_privilege_with_fallback(privileges.DATA_DICTIONARY)
+def data_dictionary_json_case_types(request, domain):
+    fhir_resource_type_name_by_case_type = {}
+    if toggles.FHIR_INTEGRATION.enabled(domain):
+        fhir_resource_type_name_by_case_type = load_fhir_case_type_mapping(domain)
+
+    case_type_app_module_count = get_case_type_app_module_count(domain)
+    used_props_by_case_type = get_used_props_by_case_type(domain)
+    geo_case_prop = get_geo_case_property(domain)
+
+    queryset = CaseType.objects.filter(domain=domain).annotate(properties_count=Count('property'))
+    if not request.GET.get("load_deprecated_case_types", False) == "true":
+        queryset = queryset.filter(is_deprecated=False)
+    case_types_data = []
+    for case_type in queryset:
+        used_props = used_props_by_case_type.get(case_type.name, [])
+        case_types_data.append({
+            "name": case_type.name,
+            "fhir_resource_type": fhir_resource_type_name_by_case_type.get(case_type),
+            "is_deprecated": case_type.is_deprecated,
+            "module_count": case_type_app_module_count.get(case_type.name, 0),
+            "is_safe_to_delete": len(used_props) == 0,
+            "properties_count": case_type.properties_count,
+        })
+
+    return JsonResponse({
+        "case_types": case_types_data,
+        "geo_case_property": geo_case_prop,
+    })
+
+
+@login_and_domain_required
+@requires_privilege_with_fallback(privileges.DATA_DICTIONARY)
+def data_dictionary_json_case_properties(request, domain, case_type_name):
+    fhir_resource_type_name_by_case_type = {}
+    fhir_resource_prop_by_case_prop = {}
+    if toggles.FHIR_INTEGRATION.enabled(domain):
+        fhir_resource_type_name_by_case_type = load_fhir_case_type_mapping(domain)
+        fhir_resource_prop_by_case_prop = load_fhir_case_properties_mapping(domain)
+
+    case_type_app_module_count = get_case_type_app_module_count(domain)
+    used_props_by_case_type = get_used_props_by_case_type(domain)
+    geo_case_prop = get_geo_case_property(domain)
+
+    def _get_case_data(case_type):
+        module_count = case_type_app_module_count.get(case_type.name, 0)
+        used_props = used_props_by_case_type.get(case_type.name, [])
+        return {
+            "name": case_type.name,
+            "fhir_resource_type": fhir_resource_type_name_by_case_type.get(case_type),
+            "is_deprecated": case_type.is_deprecated,
+            "module_count": module_count,
+            "is_safe_to_delete": len(used_props) == 0,
+            "properties_count": case_type.properties_count,
+        }
+
+    try:
+        skip = int(request.GET.get('skip', 0))
+        limit = int(request.GET.get('limit', 500))
+        if skip < 0 or limit < 0:
+            raise ValueError
+    except ValueError:
+        return JsonResponse({"error": _("skip and limit must be positive integers")}, status=400)
```

Comment:
If you're raising an exception and then immediately catching it and not reraising it, I think it's okay to leave it out.
```suggestion
    skip = int(request.GET.get('skip', 0))
    limit = int(request.GET.get('limit', 500))
    if skip < 0 or limit < 0:
        return JsonResponse(
            {"error": _("skip and limit must be positive integers")},
            status=400,
        )
```

Reply:
I added the try catch block for situations where skip or limit are not integers. The raise (for negative integers) is just for reusing the line where the error is returned.

Reply:
Ah! Good thinking ... but in that case you'll need to catch `TypeError` too:
```python
>>> 'foo' < 0
TypeError: '<' not supported between instances of 'str' and 'int'
```

Reply:
That is correct. I think the below code should raise `ValueError`  for such cases before we go on to comparison.
```
skip = int(request.GET.get('skip', 0))
limit = int(request.GET.get('limit', 500))
```

Reply:
Ah! Thank you!